index.js

```diff
@@ -296,21 +312,64 @@ function createFocusTrap(element, userOptions) {
   // kind of need to capture the action at the keydown phase.
   function checkTab(e) {
     updateTabbableNodes();
-    if (e.shiftKey && e.target === state.firstTabbableNode) {
-      e.preventDefault();
-      tryFocus(state.lastTabbableNode);
-      return;
+    const isShift = e.shiftKey;
+
+    if (isShift) {
+      const startOfGroupIndex = state.tabbableGroups.findIndex(
+        ({ firstTabbableNode }) => e.target === firstTabbableNode
+      );
+
+      const isStartOfAGroup = startOfGroupIndex !== -1;
+      if (isStartOfAGroup) {
+        e.preventDefault();
+
+        const isFirstGroup = startOfGroupIndex === 0;
+        const lastGroupIndex = state.tabbableGroups.length - 1;
+        const previousGroupIndex = startOfGroupIndex - 1;
```

Comment:
So maybe just this? We're just needing to go backward one, and loop around if we're already at the start.
```javascript
const destinationGroupIndex = startOfGroupIndex === 0
  ? state.tabbableGroups.length - 1
  : startOfGroupIndex - 1;
const destinationGroup = state.tabbableGroups[destinationGroupIndex]; // we know we'll get a group
const destinationNode = destinationGroup.lastTabbableNode;
```

Reply:
You were right about not needing to check if we have the group or not.  Thanks for that.

I don't necessarily think that consolidating to fewer lines is always beneficial.  I like naming logic with variables, so that I can read the code like in english in my head.  I don't think your request is at all on another extreme, but I'm just explaining that it was intentional to have good variable names and make each statement simple to read and understand.

I made the changes however.